src/link-nodes.js

```diff
@@ -54,17 +53,12 @@ export const recordInsertBefore = (node, container, ref_node) => {
   }
   // handle document fragments
   if (node.nodeType === Node.DOCUMENT_FRAGMENT_NODE) {
-    let c$ = node[utils.SHADY_PREFIX + 'childNodes'];
-    for (let i=0; i < c$.length; i++) {
-      linkNode(c$[i], container, ref_node);
+    const first = node[utils.NATIVE_PREFIX + 'firstChild'] || null;
```

Comment:
Is `|| null` required? Native should always return `null`, no?

Reply:
Yup, removing.